File: Arcade/TheCore/Java/IsPower.java
final class IsPower {
    
    boolean isPower(int n) {
        if (n == 1) return true;
        
        for (int p = 2; p < n; p++) {
            for (int q = 2; q < n; q++) {
                if (Math.pow(p, q) > n) break;
                if (Math.pow(p, q) == n) return true;
            }
        }
        
        return false;
    }
}

File: Arcade/TheCore/Kotlin/isPower.kt
fun isPower(n: Int): Boolean {
    if (n == 1) return true;
    
    outer@ for (i: Int in (2..n)) {
        for(j: Int in (2..n)) {
            val q: Int = Math.pow(i.toDouble(), j.toDouble()).toInt();
            if (q > n) continue@outer
            if (q == n) return true
        }
    }
    
    return false;
}